tendrl/commons/flows/expand_cluster/__init__.py

```diff
@@ -26,119 +26,144 @@ def run(self):
                         flow_id=self.parameters['flow_id'],
                         priority="info",
                         publisher=NS.publisher_id,
                         payload={
-                            "message": "SSH setup completed for all "
-                            "nodes in cluster %s" % integration_id
+                            "message": "Expanding Gluster Storage"
+                            " Cluster %s" % integration_id
                         }
                     )
                 )
-
-                break
-
-        # SSH setup jobs finished above, now install sds
-        # bits and create cluster
-        if "ceph" in sds_name:
+                gluster_help.expand_gluster(self.parameters)
             Event(
                 Message(
                     job_id=self.parameters['job_id'],
-                    flow_id = self.parameters['flow_id'],
+                    flow_id=self.parameters['flow_id'],
                     priority="info",
                     publisher=NS.publisher_id,
                     payload={
-                        "message": "Expanding ceph cluster %s" % integration_id
+                        "message": "SDS install/config completed on newly expanded nodes, "
+                        "Please wait while tendrl-node-agents detect sds details on the newly expaned nodes %s" % self.parameters['Node[]']
                     }
                 )
             )
-            ceph_help.expand_cluster(self.parameters)
-        else:
+
+            # Wait till detected cluster in populated for nodes
+            while True:
+                gevent.sleep(3)
+                all_status = []
+                detected_cluster = ""
+                different_cluster_id = False
+                dc = ""
+                for node in self.parameters['Node[]']:
+                    try:
+                        dc = NS._int.client.read(
+                            "/nodes/%s/DetectedCluster/detected_cluster_id" % node
+                        ).value
+                        if not detected_cluster:
+                            detected_cluster = dc
+                        else:
+                            if detected_cluster != dc:
+                                all_status.append(False)
+                                different_cluster_id = True
+                                break
+                        all_status.append(True)
+                    except etcd.EtcdKeyNotFound:
+                        all_status.append(False)
+                if different_cluster_id:
+                    raise FlowExecutionFailedError(
+                        "Seeing different detected cluster id in"
+                        " different nodes. %s and %s" % (
+                            detected_cluster, dc)
+                    )
+
+                if all_status:
+                    if all(all_status):
+                        break
+        except Exception as ex:
```

Comment:
But @shtripat node release should happen for any error, i feel here we can't predict